### ISSUE TITLE
New feature, Model 3D format support

### ISSUE THREAD

**Issue by bztsrc** [guillaumechereau/goxel#187]
Hi,

I'd like to say this is an awesome project!

I'm about to add an "export to a new file format" function to Goxel, and I'd like to know your opinion on one thing: the format itself is simple, so it is trivial to handle, but it also has an stb-style ANSI C SDK. Which one would you prefer:
1. implement src/formats/m3d.c from ground-up (using one of the existing formats' code as example)
2. adding the SDK to ext_src and create a wrapper in src/formats/m3d.c?

For me it's equal which option you prefer, because I'll be able to import m3d files exported from Goxel. A simplified description of the file format (limited to coloured voxels) can be found here: https://gitlab.com/bztsrc/model3d/-/blob/master/docs/voxel_format.md
In a nutshell it is similar to VOX format:
VOXT chunk: there's only one of this chunk per file, similar to RGBA chunk in VOX
VOXD chunk: voxel data, this is a multiple chunk, there should be one per Goxel Layer. Unlike XYZI in VOX files, this is RLE compressed and does not store coordinates for every voxel, just one per chunk.

Cheers,
bzt

**Comment by bztsrc** [comment]
I went on the SDK-less route, submitted the pull request.

Cheers,
bzt
